pkg/controller/workload/reconciler.go

```diff
@@ -0,0 +1,26 @@
+package workload
+
+import (
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// Reconciler acts as a client and state holder of the controller
+// for use within the rconciliation loop.
+type Reconciler interface {
+	client.Client
```

Comment:
Tiny nit, but I think it looks cleaner to have a space between 13-14 (the interface and the block of methods)